### PR TITLE
Readme of game execution.

### PR DESCRIPTION
**Pull request by Eirikalb**
# Summary

Adds documentation for the order of execution in the game

## Additional Information

Provides overview and some detail about functions and where to find them in the repo. The entries are ordered consecutively in terms of execution order.

## Test Plan
No testing, its a readme....

### CODE REVIEW

**Comment by QFSW** [comment]
# ⚠️Warning! This diff contains changes in protected folders

File: ExecutionOrder.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+## Initialisations
+
+| Filename | Function | Description |
+| ---- | ---- | ---- |
+| main.go |  main | Begins game |
+| internal/server/server.go | EntryPoint| Creates deep copy of list of gamestates. Then, while game is not over, calls runTurn and keeps track of the states during the game. |
+|internal/server/turn.go|gameOver| Checks at least one client is alive and we haven't reached maximum number of turns or seasons. |
+|internal/server/turn.go |runTurn| Gets a start of turn update, runs the organisations and runs the end of turn procedures. |
+|internal/server/turn.go|startOfTurnUpdate| Sends update of gameState to alive clients. |
```
Comment (lhl2617) [inline ExecutionOrder.md]: changed to startOfTurn

**Review by Darrekt — COMMENTED** [review]
A welcome addition :)

File: ExecutionOrder.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+|internal/server/iigointernal/base_judge.go|declarePresidentPerformanceWrapped| Uses DelcarePresidentPerformance (overwrite for non-MVP for own implementation) to evaluate president performance and broadcast result to all islands. |
+|internal/server/iigointernal/base_judge.go| declareSpeakerPerformanceWrapped| Uses DeclareSpeakerPerformance (overwrite for non-MVP for own implementation) to evaluate speaker performance and broadcast result to all islands. |
+|internal/server/iigointernal/base_speaker.go|appointNextJudge|Subtract 10 as cost of action. Propose and carry out election of next judge and returns results. * |
+|internal/server/iifo.go|runIIFO|Runs prediction session. |
+|internal/server/iifo.go|runPredictionSession| Get islands' predictions and distribute predictions to all the islands. |
+|internal/server/iifo.go|getPredictions|For alive clients, get their prediction for disaster (and foraging - non-MVP) using MakePrediction which is implemented by each client.|
+|internal/server/iifo.go|distributePreictions|Share predictions with islands of your choice. Islands who receive a prediction(s) do not know who else receieved the same prediciton(s).|
```
Comment (NGovani) [inline ExecutionOrder.md]: distributePredictions*

File: ExecutionOrder.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+|internal/server/iito.go|runIITO|MVP: gifts session only.|
+|internal/server/iito.go|runGiftSession|Gets all fiftr requests and offers. Gets the result (accept/reject gift(s)) and distributes gifts accordingly. |
+|internal/server/iito.go|getGiftAcceptance|Matches gift requests and offers per client, then allows clients to accept/reject offers. Then, returns result (accept/reject) of offer/request pairs. |
+|internal/server/turn.go|endOfTurn|Runs the organisations end of turn, forage, rpobe for disasters, increase turn and season count, deduct cost of living from clients and check clients' alive/dead status. |
+|internal/server/turn.go|runOrgsEndOfTurn|Runs end of turn for IIGO then IIFO then IITO.|
+|internal/server/iigo.go|runIIGOEndOfTurn|Get tax from each alive client and add it to the common pool. Update the game state. Logs how much tax each client contributed. |
+|internal/server/iifo.go|runIIFOEndOfTurn| To be implemented. |
+|internal/server/iito.go|runIITOEndOfTurn| To be implemented. |
```
Comment (NGovani) [inline ExecutionOrder.md]: Now updates the server state as well as notifies clients

File: ExecutionOrder.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+|internal/server/disaster.go|probeDisaster|Checks if a disaster occurs this turn. |
+|internal/server/turn.go|incrementTurnAndSeason|Increments turn. Increments season if a disaster occurs.|
+|internal/server/turn.go|deductCostOfLiving|Deduct cost of living from each alive client then update the gamestate. |
+|internal/server/turn.go|updateIslandLivingStatus|Update island status to dead is needed. |
+
+\* The same process is repeated for the other roles .
+
+## Questions
+
+- internal/server/iigointernal/orchestration.go - what happens if president and judge can be paid but speaker can't be? How are the salaries that are taken from the pool and stored returned back to the common pool? IIGO is exited before they can do their job.
```
Comment (NGovani) [inline ExecutionOrder.md]: @Neelesh99 Correct me if im wrong, but if any of the three roles cannot be paid then all salaries are given back to the common pool in the returnWithdrawnSalariesToCommonPool function.

**Review by NGovani — CHANGES_REQUESTED** [review]
Once PR #173 gets merged in, this will need some overhauls. Could we wait for that then update this PR

**Comment by lhl2617** [comment]
@NGovani passing this on to you bc you are most familiar

**Comment by NGovani** [comment]
pr #251 will replace this

**Pull request closed without merging**